File: Java/twosum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> numToIndex = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];

            if (numToIndex.containsKey(difference)) {
                return new int[] { numToIndex.get(difference), i };
            }

            numToIndex.put(nums[i], i);
        }

        return new int[] {};
    }
}

File: Javascript/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numToIndex = {};

    for (let i = 0; i < nums.length; i++) {
        const difference = target - nums[i];

        if (difference in numToIndex) {
            return [numToIndex[difference], i];
        }

        numToIndex[nums[i]] = i;
    }

    return [];
};
const target = 9;
const x = [2,7,11,15];
console.log(twoSum(x,target));